### PR TITLE
Implemented GetTransactionByBlockNumber/HashAndIndex, GetBlockTransactionCountByHash

### PR DESCRIPTION
**Pull request by novosandara**
# Description

Implemented methods: eth_getTransactionByBlockNumberAndIndex, eth_getTransactionByBlockHashAndIndex, eth_getBlockTransactionCountByHash.

# Changes include

- [ ] Bugfix (non-breaking change that solves an issue)
- [ ] Hotfix (change that solves an urgent issue, and requires immediate attention)
- [x] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (change that is not backwards-compatible and/or changes current functionality)

# Checklist

- [x] I have assigned this PR to myself
- [x] I have added at least 1 reviewer
- [ ] I have added the relevant labels
- [ ] I have updated the official documentation
- [x] I have added sufficient documentation in code

## Testing

- [x] I have tested this code with the official test suite
- [x] I have tested this code manually

### Manual tests

Please complete this section if you ran manual tests for this functionality, otherwise delete it

# Documentation update

Please link the documentation update PR in this section if it's present, otherwise delete it

# Additional comments

Please post additional comments in this section if you have them, otherwise delete it

### CODE REVIEW

**Comment by novosandara** [comment]
I have read the CLA Document and I hereby sign the CLA

File: jsonrpc/eth_endpoint.go [hunk 1]
```diff
@@ -191,6 +202,57 @@ func (e *Eth) GetBlockTransactionCountByNumber(number BlockNumber) (interface{},
...
 
+// GetTransactionByBlockNumberAndIndex returns the transaction for the given block number and index.
+func (e *Eth) GetTransactionByBlockNumberAndIndex(number BlockNumber, index argUint64) (interface{}, error) {
+	num, err := GetNumericBlockNumber(number, e.store)
+	if err != nil {
+		return nil, err
+	}
+
+	block, ok := e.store.GetBlockByNumber(num, true)
+
```
Comment (goran-ethernal) [inline jsonrpc/eth_endpoint.go]: Nitpick: remove this empty line.

**Review by goran-ethernal — DISMISSED** [review]
LGTM, just remember to fix the Lint errors.

**Review by Stefan-Ethernal — DISMISSED** [review]

File: jsonrpc/eth_endpoint.go (hunk 1 above)
Comment (Stefan-Ethernal) [reply jsonrpc/eth_endpoint.go]: Fixed https://github.com/Ethernal-Tech/blade/pull/144/commits/c758e98c14db22b834fc8504af9c752cbfcec46b

**Review by goran-ethernal — DISMISSED** [review]

**Review by Stefan-Ethernal — APPROVED** [review]

**Review by goran-ethernal — APPROVED** [review]

**Pull request merged**